returnn/util/file_cache.py

```diff
@@ -19,9 +19,59 @@
 from contextlib import contextmanager
 from threading import Thread, Event
 from .basic import expand_env_vars, LockFile, human_bytes_size
-
-
-__all__ = ["FileCache", "CachedFile"]
+from returnn.config import Config, get_global_config
+
+
+__all__ = ["FileCache", "CachedFile", "get_instance", "init_by_config"]
+
+
+DEFAULT_CACHE_DIR = "$TMPDIR/$USER/returnn/file_cache"
+DEFAULT_CLEANUP_FILES_ALWAYS_OLDER_THAN_DAYS = 31.0
+DEFAULT_CLEANUP_FILES_WANTED_OLDER_THAN_DAYS = 7.0
+DEFAULT_CLEANUP_DISK_USAGE_WANTED_FREE_RATIO = 0.2  # try to free at least 20% disk space
+DEFAULT_NUM_TRIES = 3  # retry twice by default
+
+
+cache_instance: Optional["FileCache"] = None
+
+
+def init_by_config(config: Config):
+    """Initializes the global file cache using values from `config`."""
+    global cache_instance
+    if cache_instance:
+        return
+    # TODO: just pass along all `file_cache_*` config values as kwargs?
```

Comment:
Yes of course. There should not be single config option for every single arg, but instead just sth like `file_cache_opts`.